content/blog/2024-05-23-linebender-roadmap.md

```diff
@@ -0,0 +1,248 @@
+- **Assigned:** Olivier Faure.
+- **Description:** Xilem's lack of documentation is one of the problems new users complain about the most often. We need to avoid staying in that state for too long.
+- **Issue:** [xilem#392](https://github.com/linebender/xilem/issues/392)
+
+### Re-implement async integration
+
+- **Priority:** Medium.
+- **Assigned:** Daniel McNab.
+- **Description:** One of Xilem's selling point is the ability to handle virtual lists and async sources of data. That ability was temporarily removed for the Masonry port, but we'd like to add it back it. We don't want Masonry to be aware of async stuff, but the AppDriver trait should at least provide a method which lets Xilem access Masonry internals on its own initiative.
+- **Issue:** [xilem#340](https://github.com/linebender/xilem/issues/340)
+
+### Add unit tests
+
+- **Priority:** Low.
+- **Assigned:** Olivier Faure.
+- **Description:** We need to figure out an architecture for doing unit tests in Xilem. Tests can then be added by volunteer contributors.
+- **Issue:** [xilem#393](https://github.com/linebender/xilem/issues/393)
+
+### Add basic benchmarks
+
+- **Priority:** Low.
+- **Assigned:** Olivier.
+- **Description:** Right now we're mostly working off vibes to know whether or not any of our work improved performance or introduced regressions.
+We should write at least *one* benchmark that would test a basic Xilem use-case, so we get a general idea how our performance is progressing.
+- **Issue:** [xilem#362](https://github.com/linebender/xilem/issues/362)
+
+
+## AccessKit
+
+Matt Campbell's contract with Google hasn't started yet.
+When it does, top priority will likely be the Android backend.
+
+The plan is to study existing Android accessibility implementations for non-Java toolkits, including Flutter, Chromium, and Gio, to get a better understanding of what's required to implement Android accessibility from scratch.
+
+Beyond Android support, we've only had cursory discussions of what work Matt will do for the rest of the year.
+Possible themes include scrollable containers, virtualized lists, richer text support (e.g. exposing font and style attributes) across platforms, and better integration into Xilem/Masonry.
+
+
+## Android-specific work
+
+Getting the Android port right involves a lot of small changes and making things feel non-broken.
+
+Some notable sub-goals:
+
+### Subclassing View/Application/Activity
+
+- **Priority:** Highest.
+- **Assigned:** Aaron Muir Hamilton, Raph Levien.
+- **Description:** The Java classes of an Android application are where the rubber meets the road. We're investigating what we want our boilerplate to look like.
+
+### Handling IME
+
+- **Priority:** Highest.
+- **Assigned:** Aaron Muir Hamilton, Raph Levien.
+- **Description:** We need to extend Winit to be compatible with Android IME interfaces. An Android application without IME is a non-starter. Our design choices here will have influence on accessibility as well.
```

Comment:
IME is very jargonny here; I think we should also mention virtual keyboards here, so that someone without domain knowledge can understand this item. I.e.
```suggestion
- **Description:** We need to extend Winit to be compatible with Android IME interfaces. An Android application without support for virtual keyboard and other input method is a non-starter. Our design choices here will have influence on accessibility as well.
```